docs/content/customization/nutanix/machine-details.md

```diff
@@ -50,6 +50,9 @@ spec:
               subnets:
               - name: subnet-name
                 type: name
+              additionalCategories:
+              - key: example-key
+                value: example-value
```

Comment:
Assuming the ordering does not matter, what do we think about simplifying this api in CAREN to a map?
```
additionalCategories:
  example-key: example-value
```

Reply:
As long as there is no technical reason to use maps.

One problem that came to mind be that some category key is not a valid JSON key. Since a category key is a string, and a JSON key can be any string (even an escaped one), that's not a problem.

Reply:
I'd like to know what @deepakm-ntnx thinks.

Reply:
We talked offline. @deepakm-ntnx made a great observation: users must be able to define multiple values for the same _category key_, which would not be possible if we use JSON keys for category keys, because the JSON keys must be unique.

Reply:
Why not map[string][]string? Having the same key multiple times looks a bit odd to me.

Reply:
sure, map[string][]string also makes sense but we need to be careful as to what is an acceptable value for key in json map vs in category keys. as long as that matches we can do whatever is shorter. eg i can create a category with key -1234 (please dont ask why but allowed thru categories UI) and value as xyz. but per https://google.github.io/styleguide/jsoncstyleguide.xml?showone=Property_Name_Format#Property_Name_Format this is not a valid key format. hence existing way makes more sense

Reply:
Thanks for the discussion, though!

Reply:
I experimented a little bit more. As far as I can tell, the set of JSON strings is a subset of Nutanix category keys.

So, in theory, we can support something like
```
additionalCategories:
  "-1234":
    - value1
    - value2
  anotherkey:
    - value1
    - value2
```

That's shorter than:

```
additionalCategories:
  - key: "-1234"
    value: value1
  - key: "-1234"
    value: value2
  - key: anotherkey
    value: value1
  - key: anotherkey
    value: value2
```

On the other hand, explicit fields might be easier to understand.